### PR TITLE
Clean up the embedder API and refactor

### PR DESCRIPTION
**Pull request by swift-kim**
Clean up the Tizen embedder API and refactor some parts of the embedder.

APIs changed:
- Change the API prefix `Flutter` to `FlutterDesktop` (this has been discussed once before somewhere in the internal repo but I'd like to change it back to more verbose one to avoid ambiguity and for consistency with other plugin/texture APIs)
  - Why is it ambiguous? e.g. It's hard to tell whether `FlutterEngineProperties` is a Flutter embedder API (`embedder.h`) or Tizen API (`flutter_tizen.h`).
- Added: `FlutterDesktopEngineGetMessenger` (for later use)
- Removed: `FlutterCreateWindow`, `FlutterDestroyWindow`, `FlutterRunsPrecompiledCode`
- `FlutterWindowControllerState` is replaced by `FlutterDesktopEngineRef` (because the name is not appropriate for headless mode)
- `FlutterDesktopRunEngine` takes a `bool` parameter

Other changes:
- Rename `TizenEmbedderEngine` to `FlutterTizenEngine` (which is similar to `FlutterWindowsEngine`) and clean up
- Use consistent names for fields and functions (e.g. `renderer` instead of `tizen_renderer`)
- Rename variables and constants according to naming convention
- Remove unnecessary logs

A follow-up PR will be made in the tool repo soon.

More refactoring should be done in the future. I mainly focused on cleaning up `flutter_tizen.cc` and `flutter_tizen_engine.h` in this PR.

### CODE REVIEW

File: shell/platform/tizen/channels/localization_channel.cc [hunk 1]
```diff
@@ -73,8 +73,8 @@ void LocalizationChannel::SendPlatformResolvedLocale() {
     return;
   }
 
-  FlutterLocale* flutterLocale = GetFlutterLocale(locale);
-  if (!flutterLocale) {
+  FlutterLocale* flutter_locale = GetFlutterLocale(locale);
```
Comment (bbrto21) [inline shell/platform/tizen/channels/localization_channel.cc:76]: This makes me feel comfortable :smiley: thank you!

File: shell/platform/tizen/flutter_tizen.cc [hunk 2]
```diff
@@ -10,53 +10,37 @@
...
-
-  auto state = std::make_unique<FlutterWindowControllerState>();
-  state->engine = std::make_unique<TizenEmbedderEngine>();
-
-  if (!state->engine->RunEngine(engine_properties)) {
-    FT_LOGE("Failed to run the Flutter engine.");
-    return nullptr;
-  }
+// Returns the engine corresponding to the given opaque API handle.
+static FlutterTizenEngine* EngineFromHandle(FlutterDesktopEngineRef ref) {
```
Comment (bbrto21) [inline shell/platform/tizen/flutter_tizen.cc:19]: I'm used to the expression below, what do you think?
```suggestion
// Returns the engine corresponding to the given opaque API handle.
static FlutterTizenEngine* ToImpl(FlutterDesktopEngineRef ref) {
```

File: shell/platform/tizen/flutter_tizen.cc [hunk 3]
```diff
@@ -10,53 +10,37 @@
...
+  return reinterpret_cast<FlutterTizenEngine*>(ref);
+}
 
-  return state.release();
+// Returns the opaque API handle for the given engine instance.
+static FlutterDesktopEngineRef HandleForEngine(FlutterTizenEngine* engine) {
```
Comment (bbrto21) [inline shell/platform/tizen/flutter_tizen.cc:24]: It's a pair of opinions with ```ToImpl```.
```suggestion
static FlutterDesktopEngineRef ToRef(FlutterTizenEngine* engine) {
```

**Review by bbrto21 — APPROVED** [review]
Generally, looks good to me!

File: shell/platform/tizen/flutter_tizen.cc (hunk 2 above)
Comment (swift-kim) [reply shell/platform/tizen/flutter_tizen.cc:19]: The name is from the windows implementation:

https://github.com/flutter-tizen/engine/blob/ad1fd5badb74632ecc88343101ff2f601574e5b4/shell/platform/windows/flutter_windows.cc#L30-L40

If `ToImpl` feels comfortable for you, I can change to it.

Comment (bbrto21) [reply shell/platform/tizen/flutter_tizen.cc:19]: Oh, I didn't know. I think the way I'm used to it is more intuitive, but no problem. 
Let's leave it as it is for the unification of terms.

**Review by bwikbs — APPROVED** [review]
I love this work. thanks :+1:

**Review by xiaowei-guan — APPROVED** [review]
LGTM

**Pull request merged**